src/data_provider/src/sysInfoLinux.cpp

```diff
@@ -355,6 +364,118 @@ nlohmann::json SysInfo::getNetworks() const
     return networks;
 }
 
+std::pair<std::string, std::string> parseProcessInfo(char path[])
+{
+    // Open stat file
+    std::pair<std::string, std::string> processInfo;
```

Comment:
You could avoid creating this variable by doing the following:

```c++
std::pair<std::string, std::string> parseProcessInfo(char path[])
{
    // Open stat file
    std::ifstream file(path, std::ios::in | std::ios::binary);
    std::string statContent;

    // Get stat file content
    std::getline(file, statContent);

    // Parse PID and process name
    const int openParentesysPos = std::string(statContent).find("(");
    const int closeParentesysPos = std::string(statContent).find(")");
    const int spacePos = std::string(statContent).find(" ");

    file.close();

    return {std::string(statContent).substr(0, spacePos), 
            std::string(statContent).substr(openParentesysPos + 1, closeParentesysPos - openParentesysPos - 1)};
}
```

Reply:
I added a check as requested for the file variable, keeping the variable to return default initialization.